website/docs/utilities/dialog-primitive/index.md

```diff
@@ -0,0 +1,28 @@
+---
+title: DialogPrimitive
+description: Dialog Primitives are the foundational blocks of dialog components such as modals, flyouts, and similar dialog-based components.
+caption: The foundational blocks of dialog components.
+related: ['components/modal', 'components/flyout']
+previewImage: assets/illustrations/utilities/dialog-primitive.jpg
```

Comment:
[blocker] this illustration is missing

Reply:
Yep, I'll take a stab at it

Reply:
Added an image in https://github.com/hashicorp/design-system/pull/1945/commits/577d499252684858d6856f1919ad86789e82c1d9

Reply:
Thanks!